sdformat_urdf/CMakeLists.txt

```diff
@@ -14,18 +14,60 @@ find_package(ament_cmake_ros REQUIRED)
 
 find_package(pluginlib REQUIRED)
 find_package(rcutils REQUIRED)
-find_package(sdformat9 REQUIRED)
 find_package(urdfdom_headers REQUIRED)
 find_package(urdf_parser_plugin REQUIRED)
 find_package(tinyxml2_vendor REQUIRED)
 find_package(TinyXML2 REQUIRED)
 
+# Choose SDF version
+
+# Default to Citadel
+set(SDF_VER 9)
+
+# First try GZ_VERSION
+
+# Edifice
+if("$ENV{GZ_VERSION}" STREQUAL "edifice")
+  find_package(sdformat11 REQUIRED)
+  set(SDF_VER ${sdformat11_VERSION_MAJOR})
+
+  message(STATUS "Compiling against Gazebo Edifice (SDF 11), requested with GZ_VERSION")
+# Fortress
+elseif("$ENV{GZ_VERSION}" STREQUAL "fortress")
+  find_package(sdformat12 REQUIRED)
+  set(SDF_VER ${sdformat12_VERSION_MAJOR})
+
+  message(STATUS "Compiling against Gazebo Fortress (SDF 12), requested with GZ_VERSION")
+elseif("$ENV{GZ_VERSION}" STREQUAL "citadel")
+  find_package(sdformat9 REQUIRED)
+  set(SDF_VER ${sdformat9_VERSION_MAJOR})
+
+  message(STATUS "Compiling against Gazebo Citadel (SDF 9), requested with GZ_VERSION")
+endif()
+# Then try ROS_DISTRO
+elseif("$ENV{ROS_DISTRO}" STREQUAL "galactic")
```

Comment:
I'd recommend using `ROS_DISTRO` at the top to remove the duplicate `find_package(sdformat11)`.


Something like

```cmake
# If the user didn't specify a GZ distribution, pick the one matching the ROS distribution according to REP 2000
if(NOT DEFINED ENV{GZ_VERSION} AND DEFINED ENV{ROS_DISTRO})
    if("$ENV{ROS_DISTRO}" STREQUAL"galactic")
        set(ENV{GZ_VERSION} "fortress")
    elseif("$ENV{ROS_DISTRO}" STREQUAL "humble")
        # ...
    endif()
endif()

# Find libsdformat matching the picked GZ distribution
if("$ENV{GZ_VERSION}" STREQUAL "fortress")
    find_package(sdformat12 REQUIRED)
    # ...
elif("$ENV{GZ_VERSION}" STREQUAL ...)
else()
# ... No GZ distribution specified, find any version of libsdformat we can
endif()

Reply:
Cool, done in [55e0b38](https://github.com/ros/sdformat_urdf/pull/8/commits/55e0b38af9e2b65218ab683133b80993e4308f1e) and besides `humble` also added `foxy` for completeness (although I believe that may never work, see #3 )